lib/ansible/vars/manager.py

```diff
@@ -345,7 +345,12 @@ def plugins_by_groups():
         if play:
             all_vars = combine_vars(all_vars, play.get_vars())
 
-            for vars_file_item in play.get_vars_files():
+            vars_files = play.get_vars_files()
+            if not hasattr(vars_files, '__iter__'):
```

Comment:
bikeshed: its normally cheaper to catch the exception vs testing for attribute every time

Reply:
Yeah, I just wasn't sure if we wanted to nest the whole loop in a try/except. Would that be preferable to having this check?

Reply:
yeah, as long as you are capturing the specific exception

Reply:
Done